libs/s25main/mapGenerator/Resources.cpp

```diff
@@ -71,10 +72,22 @@ namespace rttr { namespace mapGenerator {
                    || helpers::contains_if(map.harbors, [pt](const Triangle& tr) { return tr.position == pt; });
         };
 
-        auto isForbiddenArea = [&isForbidden, &map](const MapPoint& pt) {
-            return helpers::contains_if(map.textures.GetPointsInRadiusWithCenter(pt, 5), isForbidden)
-                   || map.textureMap.Any(pt, IsSnowOrLava);
-        };
+        RTTR_FOREACH_PT(MapPoint, map.size)
+        {
+            if(isForbidden(pt))
+            {
+                auto suroundingArea = map.textures.GetPointsInRadiusWithCenter(pt, 5);
+                for(const auto& point : suroundingArea)
+                {
+                    forbiddenArea.insert(point);
+                }
+            } else if(map.textureMap.Any(pt, IsSnowOrLava))
+            {
+                forbiddenArea.insert(pt);
+            }
+        }
+
+        auto isForbiddenArea = [&forbiddenArea](const MapPoint& pt) { return helpers::contains(forbiddenArea, pt); };
```

Comment:
As mentioned earlier: Why not pass this collection (can stay a set) into `Distances` directly? This would greatly improve performance as it saves a whole iteration and lot's of contains checks (although `set` makes them quite cheap)